code/game/machinery/vending_types.dm

```diff
@@ -715,8 +739,37 @@
 	name = "container dispenser"
 	desc = "A container that dispenses containers."
 	icon_state = "robotics"
+	vend_id = "baygay"
 	products = list(
 		/obj/structure/closet/crate/freezer = 2,
 		/obj/structure/closet = 3,
 		/obj/structure/closet/crate = 3
 	)
+
+
+//RECURSION
+/obj/machinery/vending/vendors
+	name = "Omni-Vendor"
+	desc = "The mother of all vendors, from which vending itself comes!"
+	icon_state = "engivend"
+	icon_deny = "engivend-deny"
+	vend_id = "tools"
```

Comment:
That vend id is wrong.
It should be "restocking unit" and not "tools".

But as I mentioned above. This vendor should not be restockable with a refill cartridge or the cartridge should not be available normally in game -> Not even as a premium item